src/nghttp.cc

```diff
@@ -304,6 +305,52 @@ void Request::record_response_end_time() {
 }
 
 namespace {
+void continue_timeout_cb(struct ev_loop *loop, ev_timer *w, int revents) {
+  auto client = static_cast<HttpClient *>(ev_userdata(loop));
+  auto req = static_cast<Request *>(w->data);
+  int error;
+
+  error = nghttp2_submit_data(client->session, NGHTTP2_FLAG_END_STREAM,
+                              req->stream_id, req->data_prd);
+
+  if (error) {
+    std::cerr << "[ERROR] nghttp2_submit_data() returned error: "
+              << nghttp2_strerror(error) << std::endl;
+    nghttp2_submit_rst_stream(client->session, NGHTTP2_FLAG_NONE,
+                              req->stream_id, NGHTTP2_INTERNAL_ERROR);
+  }
+
+  client->signal_write();
+}
+} // namespace
+
+ContinueTimer::ContinueTimer(struct ev_loop *loop, Request *req)
+    : loop(loop),
+      req(req) {
+  ev_timer_init(&timer, continue_timeout_cb, 1., 0.);
+}
+
+ContinueTimer::~ContinueTimer() {
+  stop();
+}
+
+void ContinueTimer::start() {
+  timer.data = req;
+  ev_timer_start(loop, &timer);
+}
+
+void ContinueTimer::stop() {
+  ev_timer_stop(loop, &timer);
+}
+
+void ContinueTimer::dispatch_continue() {
+  // Only dispatch the timeout callback if it hasn't already been called.
+  if (ev_is_active(&timer)) {
+    ev_feed_event(loop, &timer, 0);
```

Comment:
It would be easier to directly call callback function here instead of ev_feed_event().